### DISCUSSION TITLE
Addresses returning empty Vec

### DISCUSSION THREAD

**Discussion by egithinji** [rust-pcap/pcap#236] [Q&A] [answered]
Hi! I'm attempting to retrieve the IP address of a device. However the following returns an emtpy vec:
`let mut handle = Device::lookup().unwrap();`
`handle.addresses`
I am able however to get the device's name using _handle.name_.

I was wondering is there a way to retrieve the IP address associated with the device if it's not the above way? Thanks

**Comment by Stargateur** [comment, marked as answer]
I'm not sure I fully understand your question, if addresses is empty it's probably mean that this devices don't have any address, you can use https://docs.rs/pcap/0.9.2/pcap/struct.Device.html#method.list and show all devices and their addresses. You may find what you search.

**Reply by egithinji** [reply]
Thanks. Using the _list()_ method has allowed me to retrieve the device's address.
The following still returns an empty vec for the _addresses_ field:
`let mut handle = Device::lookup().unwrap();`
`handle.addresses`
However this does return an IP address (and it's the same device that is retrieved):
`let mut handle = &Device::list().unwrap()[0];`
`handle.addresses`

But the _list()_ method is perfectly fine for me and has solved my problem, thanks.
